### ISSUE TITLE
(codebuild): change from with vpc to be without vpc does not remove vpc from build configuration

### ISSUE THREAD

**Issue by phsiao** [aws/aws-cdk#22733]
### Describe the bug

If you build a CodeBuild Project first with a vpc and deploy, then remove the vpc and perform the second deploy, the related IAM permission would be removed but the vpc remains in the CodeBuild Project's configuration.

Below is a simple example that I can use to reproduce the problem

```
export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    const vpc = ec2.Vpc.fromLookup(this, "Vpc", {
      vpcId: "vpc-12345678",   // any valid vpc id in the target account
    });

    new codebuild.Project(this, "Project", {
      // vpc,
      buildSpec: codebuild.BuildSpec.fromObject({
        version: "0.2",
        env: {},
        phases: {
          build: {
            commands: 'echo "Hello World!"',
          },
        },
      }),
    });
  }
}
```

This is problematic especially with pipeline, because the pipeline would fail because it can't perform `ec2:DescribeNetworkInterfaces` on the vpc and the project fails during Queue time.

### Expected Behavior

The VPC configuration be cleared from the CodeBuild Project.

### Current Behavior

The VPC configuration remains in the CodeBuild Project according to console and cli.

### Reproduction Steps

First deploy this

```
export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    const vpc = ec2.Vpc.fromLookup(this, "Vpc", {
      vpcId: "vpc-12345678",   // any valid vpc id in the target account
    });

    new codebuild.Project(this, "Project", {
      vpc,
      buildSpec: codebuild.BuildSpec.fromObject({
        version: "0.2",
        env: {},
        phases: {
          build: {
            commands: 'echo "Hello World!"',
          },
        },
      }),
    });
  }
}
```

then deploy this

```
export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    ec2.Vpc.fromLookup(this, "Vpc", {
      vpcId: "vpc-12345678",   // any valid vpc id in the target account
    });

    new codebuild.Project(this, "Project", {
      // vpc,
      buildSpec: codebuild.BuildSpec.fromObject({
        version: "0.2",
        env: {},
        phases: {
          build: {
            commands: 'echo "Hello World!"',
          },
        },
      }),
    });
  }
}
```

The second deploy would remove the Project role's ability to perform various ec2 actions, including `ec2:DescribeNetworkInterfaces`.  But the vpc configuration remains in the Project (according to console as well as aws cli)

### CDK CLI Version

2.50.0

### Node.js Version

v18.6.0

### OS

MacOS

### Language

Typescript

**Comment by peterwoodworth** [comment]
Thanks for reporting this @phsiao, I've been able to confirm this behavior.

This appears to be a bug on CloudFormation's end - the template we deploy is accurate and doesn't include any Vpc configurations when not defined, but the infrastructure isn't updated when CloudFormation updates the resource. 

I've reported this internally to CloudFormation - P74606487 - I will provide updates as they become available

**Comment by peterwoodworth** [comment]
OK they've let us know that to actually unset the VPC we need to pass in an empty object (see below), otherwise it will maintain the same value.

```yaml
Resources:
  CodeBuildProject:
    Type: AWS::CodeBuild::Project
    Properties:
      Name: aaa
      Description: testss
      ServiceRole: somerole
        Type: CODEPIPELINE
      Environment:
        Type: LINUX_CONTAINER
        ComputeType: BUILD_GENERAL1_SMALL
        Image: aws/codebuild/ubuntu-base:14.04
      Source:
        Type: CODEPIPELINE
      TimeoutInMinutes: 10
      Tags: []
      VpcConfig: {}
```

Not only is this true for `VpcConfig` property, it's true for all CodeBuild Project properties

> per CodeBuild team, while updating a CodeBuild project, not specifying a property is equivalent to not updating it and retaining the existing value for it.

We should ensure that we account for this on all properties, not just `VpcConfig`. There are quite a few... https://github.com/aws/aws-cdk/blob/4c11af6067b35125781aa590bb33c7990078d1ed/packages/%40aws-cdk/aws-codebuild/lib/project.ts#L1069

**Comment by hoegertn** [comment]
Sorry, but imho this is still a CFN bug. The behavior of CFN is to use default values when not specifying a value and not the previous value. Otherwise this would lead to two different states if you apply the same template to an existing stack vs a new stack.

Please escalate this. I am currently looking into another strange effect on the CodeBuild project resource on CFN updates. So maybe there are some more hidden bugs out there.

**Comment by peterwoodworth** [comment]
@hoegertn thanks for sharing your opinion, I've shared this concern with the team.

I've come across this behavior before as well, see https://github.com/aws/aws-cdk/issues/18077

**Comment by rix0rrr** [comment]
Yeah, CodeBuild Projects implement updates using the `UpdateProject` API, which ignores unspecified values. This is an upstream bug.

**Comment by amcubic** [comment]
In addition to not removing the VPC it also appears that the values are not updated. For example, if you change the subnets on the VPC object it does not seem to be applied. My workaround was to try and remove the VPC config and add it back though that didn't work either.

**Comment by alessandroscarlatti** [comment]
`.addPropertyOverride('VpcConfig', {});` did not work for me. The CDK optimized this away, resulting in no `VpcConfig` property being set at all. What did work for me was:

```
.addPropertyOverride('VpcConfig', {
  SecurityGroupIds: Aws.NO_VALUE,
  Subnets: Aws.NO_VALUE,
  VpcId: Aws.NO_VALUE
});
```
